app/Services/Patient/PatientRegistrationService.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using backend.Database;
+using backend.Dto.Requests.Patient;
+using backend.Exceptions;
+using backend.Helpers;
+using backend.Models.Accounts;
+using backend.Models.Accounts.AdditionalData;
+
+namespace backend.Services.Patient
+{
+	public class PatientRegistrationService
+	{
+        private readonly DataContext dataContext;
+        private readonly SecurePasswordHasher securePasswordHasher;
+
+        public PatientRegistrationService(DataContext dataContext, SecurePasswordHasher securePasswordHasher)
+        {
+            this.dataContext = dataContext;
+            this.securePasswordHasher = securePasswordHasher;
+        }
+
+        public async Task<PatientModel> Register(PatientRegistrationRequest request)
+        {
+            // Check for duplicated e-mail
+            var existingEmail = this.dataContext.Patients
```

Comment:
Może byśmy dodali extension do IEnumerate z metodą FirstOrThrow? Bo moglibyśmy skrócić ten kod a czesto używamy takiego rozwiazania ze jak null to cos rzucamy.

https://stackoverflow.com/questions/6193505/design-choice-do-i-want-extension-methods-to-throw-exceptions-on-null

Reply:
Świetny pomysł! Napisałem też extension do odwrotnej funkcjonalności, tj. sprawdzania, czy nie występuje zduplikowany element.